libtiledbsoma/test/unit_soma_group.cc

```diff
@@ -0,0 +1,226 @@
+    }
+
+    // Open array for writing
+    Array array(ctx, uri, TILEDB_WRITE, timestamp);
+    if (LOG_DEBUG_ENABLED()) {
+        array.schema().dump();
+    }
+
+    // Generate fragments in random order
+    std::vector<int> frags(num_fragments);
+    std::iota(frags.begin(), frags.end(), 0);
+    std::shuffle(frags.begin(), frags.end(), std::random_device{});
+
+    for (auto i : frags) {
+        std::vector<int64_t> d0(num_cells_per_fragment);
+        for (int j = 0; j < num_cells_per_fragment; j++) {
+            // Overlap odd fragments when generating overlaps
+            if (overlap && i & 1) {
+                d0[j] = j + num_cells_per_fragment * (i - 1);
+            } else {
+                d0[j] = j + num_cells_per_fragment * i;
+            }
+        }
+        std::vector<int> a0(num_cells_per_fragment, i);
+
+        // Write data to array
+        Query query(ctx, array);
+        query.set_layout(TILEDB_UNORDERED)
+            .set_data_buffer("d0", d0)
+            .set_data_buffer("a0", a0);
+        query.submit();
+    }
+
+    array.close();
+
+    uint64_t nnz = num_fragments * num_cells_per_fragment;
+
+    if (allow_duplicates) {
+        return {uri, nnz};
+    }
+
+    // Adjust nnz when overlap is enabled
+    if (overlap) {
+        nnz = (num_fragments + 1) / 2 * num_cells_per_fragment;
+    }
+
+    return {uri, nnz};
+}
+
+};  // namespace
+
+TEST_CASE("SOMAGroup: basic") {
+    auto ctx = std::make_shared<Context>();
+
+    std::string uri_main_group = "mem://main-group";
+    Group::create(*ctx, uri_main_group);
+
+    std::string uri_sub_group = "mem://sub-group";
+    Group::create(*ctx, uri_sub_group);
+
+    auto [uri_sub_array, expected_nnz] = create_array("mem://sub-array", *ctx);
+
+    auto sg = SOMAGroup::open(TILEDB_WRITE, ctx, uri_main_group, "metadata", 1);
+    sg->add_member(uri_sub_group, false, "subgroup");
+    sg->add_member(uri_sub_array, false, "subarray");
+    sg->close();
+
+    sg->open(TILEDB_READ, 1);
+    REQUIRE(sg->ctx() == ctx);
+    REQUIRE(sg->uri() == uri_main_group);
+    REQUIRE(sg->get_length() == 2);
+    std::map<std::string, std::string> expected_map{
+        {"subgroup", uri_sub_group}, {"subarray", uri_sub_array}};
+    REQUIRE(expected_map == sg->member_to_uri_mapping());
+    REQUIRE(sg->get_member("subgroup").type() == Object::Type::Group);
+    REQUIRE(sg->get_member("subarray").type() == Object::Type::Array);
+    sg->close();
+
+    sg->open(TILEDB_WRITE, 3);
+    sg->remove_member("subgroup");
+    sg->close();
+
+    sg->open(TILEDB_READ, 4);
+    REQUIRE(sg->get_length() == 1);
+    REQUIRE(sg->has_member("subgroup") == false);
+    REQUIRE(sg->has_member("subarray") == true);
+    sg->close();
+}
+
+TEST_CASE("SOMAGroup: metadata") {
+    auto ctx = std::make_shared<Context>();
+
+    std::string uri = "mem://unit-test-group";
+    Group::create(*ctx, uri);
+    auto sr = SOMAGroup::open(TILEDB_WRITE, ctx, uri, "metadata", 1);
```

Comment:
What does sr stand for?

Reply:
This should be changed. It use to stand for `SOMAReader` but now we change that to `SOMAGroup` and `SOMAArray`.